src/openrct2-ui/windows/News.cpp

```diff
@@ -33,316 +33,279 @@ enum WINDOW_NEWS_WIDGET_IDX {
+        static int32_t _scrollHeight = static_cast<int32_t>(gNewsItems.GetArchived().size()) * window_news_get_item_height();
+        return {WW, _scrollHeight};
     }
-}
-
-/**
- *
- *  rct2: 0x0066E4E8
- */
-static void window_news_paint(rct_window* w, rct_drawpixelinfo* dpi)
-{
-    WindowDrawWidgets(w, dpi);
-}
-
-/**
- *
- *  rct2: 0x0066E4EE
- */
-static void window_news_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi, int32_t scrollIndex)
-{
-    int32_t lineHeight = font_get_line_height(FontSpriteBase::SMALL);
-    int32_t itemHeight = window_news_get_item_height();
 
-    int32_t y = 0;
-    int32_t i = 0;
-    for (const auto& newsItem : gNewsItems.GetArchived())
+    void OnScrollMouseDown(int32_t scrollIndex, const ScreenCoordsXY& screenCoords) override
     {
-        if (y >= dpi->y + dpi->height)
-            break;
-        if (y + itemHeight < dpi->y)
+        int32_t itemHeight = window_news_get_item_height();
+        int32_t i = 0;
+        int32_t buttonIndex = 0;
+        auto mutableScreenCoords = screenCoords;
+        for (const auto& newsItem : gNewsItems.GetArchived())
         {
-            y += itemHeight;
+            if (mutableScreenCoords.y < itemHeight)
+            {
+                if (newsItem.HasButton() || mutableScreenCoords.y < 14 || mutableScreenCoords.y >= 38
+                    || mutableScreenCoords.x < 328)
+                {
+                    buttonIndex = 0;
+                    break;
+                }
+                else if (mutableScreenCoords.x < 351 && newsItem.TypeHasSubject())
+                {
+                    buttonIndex = 1;
+                    break;
+                }
+                else if (mutableScreenCoords.x < 376 && newsItem.TypeHasLocation())
+                {
+                    buttonIndex = 2;
+                    break;
+                }
+            }
+            mutableScreenCoords.y -= itemHeight;
             i++;
-            continue;
         }
 
-        // Background
-        gfx_fill_rect_inset(
-            dpi, { -1, y, 383, y + itemHeight - 1 }, w->colours[1], (INSET_RECT_FLAG_BORDER_INSET | INSET_RECT_FLAG_FILL_GREY));
-
-        // Date text
-        {
-            auto ft = Formatter();
-            ft.Add<rct_string_id>(DateDayNames[newsItem.Day - 1]);
-            ft.Add<rct_string_id>(DateGameMonthNames[date_get_month(newsItem.MonthYear)]);
-            DrawTextBasic(dpi, { 2, y }, STR_NEWS_DATE_FORMAT, ft, { COLOUR_WHITE, FontSpriteBase::SMALL });
-        }
-        // Item text
+        if (buttonIndex != 0)
         {
-            auto ft = Formatter();
-            ft.Add<const char*>(newsItem.Text.c_str());
-            DrawTextWrapped(dpi, { 2, y + lineHeight }, 325, STR_BOTTOM_TOOLBAR_NEWS_TEXT, ft, { FontSpriteBase::SMALL });
+            news.var_480 = i;
+            news.var_482 = buttonIndex;
+            news.var_484 = 4;
```

Comment:
Replace with private member field. Call it something like `_suspendUpdateTicks`.